tests/qos/qos_sai_base.py

```diff
@@ -277,11 +287,35 @@ def __buildTestPorts(self, request, testPortIds, testPortIps):
         dstPorts = request.config.getoption("--qos_dst_ports")
         srcPorts = request.config.getoption("--qos_src_ports")
 
-        pytest_assert(len(set(dstPorts).intersection(set(srcPorts))) == 0,
-                      "Duplicate destination and source ports '{0}'".format(set(dstPorts).intersection(set(srcPorts))))
+        if dstPorts is None:
+            if len(testPortIds) >= 4:
+                dstPorts = [0, 2, 3]
+            elif len(testPortIds) == 3:
+                dstPorts = [0, 2, 2]
+            else:
+                dstPorts = [0, 0, 0]
+
+        if srcPorts is None:
+            srcPorts = [1]
+
+        pytest_assert(len(testPortIds) >= 2, "Provide at least 2 test ports")
+        logging.debug(
+            "Test Port IDs:{} IPs:{}".format(testPortIds, testPortIps)
+        )
+        logging.debug("Test Port dst:{}, src:{}".format(dstPorts, srcPorts))
+
+        pytest_assert(
+            len(set(dstPorts).intersection(set(srcPorts))) == 0,
+            "Duplicate destination and source ports '{0}'".format(
+                set(dstPorts).intersection(set(srcPorts))
+            )
+        )
 
-        pytest_assert(len(dstPorts) == 3 and len(srcPorts) == 1,
-                      "Invalid number of ports provided, qos_dst_ports:{0}, qos_src_ports:{1}".format(len(dstPorts), len(srcPorts)))
+        pytest_assert(
+            len(dstPorts) == 3 and len(srcPorts) == 1,
```

Comment:
This check doesn't hold good anymore since we are anyways hardcoding 3 dst and 1 src port.